src/App.jsx

```diff
@@ -1,248 +1,414 @@
 
-                  <a href="#/">
-                    <span className="icon">
-                      <i data-cy="SortIcon" className="fas fa-sort" />
-                    </span>
-                  </a>
+                {query.length > 0 && (
+                <span className="icon is-right">
+                  {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+                  <button
+                    data-cy="ClearButton"
+                    type="button"
+                    className="delete"
+                    onClick={() => setQuery('')}
+                  />
                 </span>
-              </th>
-            </tr>
-          </thead>
-
-          <tbody>
-            <tr data-cy="Product">
-              <td className="has-text-weight-bold" data-cy="ProductId">
-                1
-              </td>
-
-              <td data-cy="ProductName">Milk</td>
-              <td data-cy="ProductCategory">🍺 - Drinks</td>
-
-              <td
-                data-cy="ProductUser"
-                className="has-text-link"
+                )}
+              </p>
+            </div>
+
+            <div className="panel-block is-flex-wrap-wrap">
+              <a
+                href="#/"
+                data-cy="AllCategories"
+                className={cn(
+                  'button',
+                  'is-success',
+                  'mr-6',
+                  { 'is-outlined': selectedCategoryIDs.length !== 0 },
+                )}
+                onClick={() => setSelectedCategoryIDs([])}
               >
-                Max
-              </td>
-            </tr>
-
-            <tr data-cy="Product">
-              <td className="has-text-weight-bold" data-cy="ProductId">
-                2
-              </td>
-
-              <td data-cy="ProductName">Bread</td>
-              <td data-cy="ProductCategory">🍞 - Grocery</td>
-
-              <td
-                data-cy="ProductUser"
-                className="has-text-danger"
+                All
+              </a>
+
+              {categoriesFromServer.map(category => (
+                <a
+                  key={category.id}
+                  data-cy="Category"
+                  className={cn(
+                    'button',
+                    'mr-2',
+                    'my-1',
+                    {
+                      'is-info': selectedCategoryIDs.includes(category.id),
+                    },
+                  )
+                  }
+                  href="#/"
+                  onClick={() => {
+                    if (selectedCategoryIDs.includes(category.id)) {
+                      setSelectedCategoryIDs(
+                        selectedCategoryIDs.filter(id => id !== category.id),
+                      );
+                    } else {
+                      setSelectedCategoryIDs(
+                        [...selectedCategoryIDs, category.id],
+                      );
+                    }
+                  }}
```

Comment:
It is better to put such huge logic into a separate function and use it here. Remember the rule, if logic takes more than 1-2 rows, put it into a separate function